backend/pkg/kafka/consumer_worker.go

```diff
@@ -0,0 +1,61 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudhut/kowl/backend/pkg/proto"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+)
+
+func (s *Service) startMessageWorker(ctx context.Context, isMessageOK isMessageOkFunc, jobs <-chan *kgo.Record, resultsCh chan<- *TopicMessage) {
+	for record := range jobs {
```

Comment:
This can be to be
```go
for {
  var record *kgo.Record
  var ok bool
  select {
  case <-ctx.Done():
    return
  case record, ok = <-jobs:
    if !ok {
      return
    }
  }
  ...
}
```
(or `record, ok :=`, and then move all logic for the loop into the case)
and then the select / default at the end of the loop can be removed.